File: java/src/main/java/vsdk/toolkit/common/statistics/RayTracerStatistics.java
package vsdk.toolkit.common.statistics;

public class RayTracerStatistics {
    public double totalTime;
    public long rayCount;
    public long pixelCount;

    public RayTracerStatistics() {
        totalTime = 0.0;
        rayCount = 0;
        pixelCount = 0;
    }

    public void resetCounters() {
        totalTime = 0.0;
        rayCount = 0;
        pixelCount = 0;
    }
}

File: typescript/src/main/vsdk/toolkit/common/statistics/RayTracerStatistics.ts
export class RayTracerStatistics {
  public totalTime: number;
  public rayCount: number;
  public pixelCount: number;

  public constructor() {
    this.totalTime = 0.0;
    this.rayCount = 0;
    this.pixelCount = 0;
  }

  public resetCounters(): void {
    this.totalTime = 0.0;
    this.rayCount = 0;
    this.pixelCount = 0;
  }
}
